cli/lsp/language_server.rs

```diff
@@ -1023,6 +1041,44 @@ impl Inner {
       self.lint_options = lint_options;
       self.fmt_options = fmt_options;
       self.recreate_http_client_and_dependents().await?;
+      if let Some(config_file) = self.config.maybe_config_file() {
+        if let Ok((compiler_options, _)) = config_file.to_compiler_options() {
+          if let Some(compiler_options_obj) = compiler_options.as_object() {
+            if let Some(jsx_import_source) =
+              compiler_options_obj.get("jsxImportSource")
+            {
+              if let Some(jsx_import_source) = jsx_import_source.as_str() {
+                let cache_params = lsp_custom::CacheParams {
+                  referrer: TextDocumentIdentifier {
+                    uri: config_file.specifier.clone(),
+                  },
+                  uris: vec![TextDocumentIdentifier {
+                    uri: Url::parse(&format!(
+                      "data:application/typescript;base64,{}",
+                      base64::engine::general_purpose::STANDARD.encode(
+                        format!("import '{jsx_import_source}/jsx-runtime';")
+                      )
+                    ))
+                    .unwrap(),
+                  }],
+                };
+                self
+                  .outside_lock_task_tx
+                  .send(Box::new(|ls: LanguageServer| {
+                    Box::pin(async move {
+                      if let Err(err) =
+                        ls.cache_request(Some(json!(cache_params))).await
+                      {
+                        lsp_warn!("{}", err);
+                      }
+                    })
+                  }))
+                  .unwrap();
```

Comment:
Why is the channel necessary? Couldn't we instead just call `spawn` here?

Reply:
> Why is the channel necessary? Couldn't we instead just call `spawn` here?

We need access to `LanguageServer` as opposed to `Inner`.